tools/src/bin/sha256tree-benching.rs

```diff
@@ -0,0 +1,449 @@
+        writeln!(output_clvm_cost, "{}\t{}", i, cost).unwrap();
+        samples_time_clvm.push((i as f64, duration));
+        samples_cost_clvm.push((i as f64, cost as f64));
+    }
+
+    (
+        linear_regression_of(&samples_time_native).unwrap().0,
+        linear_regression_of(&samples_time_clvm).unwrap().0,
+        linear_regression_of(&samples_cost_native).unwrap().0,
+        linear_regression_of(&samples_cost_clvm).unwrap().0,
+    )
+}
+
+fn main() {
+    let shaprogbytes = hex::decode(
+        "ff02ffff01ff02ff02ffff04ff02ffff04ff03ff80808080ffff04ffff01ff02ffff03ffff07ff0580ffff01ff0bffff0102ffff02ff02ffff04ff02ffff04ff09ff80808080ffff02ff02ffff04ff02ffff04ff0dff8080808080ffff01ff0bffff0101ff058080ff0180ff018080"
+    ).unwrap();
+
+    let mut a = Allocator::new();
+    let shaprog = node_from_bytes(&mut a, shaprogbytes.as_ref()).unwrap();
+
+    // Output files
+    let atom_native_time = BufWriter::new(File::create("atom_native.dat").unwrap());
+    let atom_clvm_time = BufWriter::new(File::create("atom_clvm.dat").unwrap());
+    let cons_native_time = BufWriter::new(File::create("cons_native.dat").unwrap());
+    let cons_clvm_time = BufWriter::new(File::create("cons_clvm.dat").unwrap());
+    let tree_native_time = BufWriter::new(File::create("tree_native.dat").unwrap());
+    let tree_clvm_time = BufWriter::new(File::create("tree_clvm.dat").unwrap());
+
+    let atom_native_cost = BufWriter::new(File::create("atom_native_cost.dat").unwrap());
+    let atom_clvm_cost = BufWriter::new(File::create("atom_clvm_cost.dat").unwrap());
+    let cons_native_cost = BufWriter::new(File::create("cons_native_cost.dat").unwrap());
+    let cons_clvm_cost = BufWriter::new(File::create("cons_clvm_cost.dat").unwrap());
+    let tree_native_cost = BufWriter::new(File::create("tree_native_cost.dat").unwrap());
+    let tree_clvm_cost = BufWriter::new(File::create("tree_clvm_cost.dat").unwrap());
+
+    let (atom_nat_t, atom_clvm_t, atom_nat_c, atom_clvm_c) = time_per_byte_for_atom(
+        &mut a,
+        shaprog,
+        atom_native_time,
+        atom_clvm_time,
+        atom_native_cost,
+        atom_clvm_cost,
+    );
+
+    let (leaf_nat_t, leaf_clvm_t, leaf_nat_c, leaf_clvm_c) = time_per_cons_for_balanced_tree(
+        &mut a,
+        shaprog,
+        atom_nat_c,
+        atom_clvm_c,
+        atom_nat_t,
+        atom_clvm_t,
+        tree_native_time,
+        tree_clvm_time,
+        tree_native_cost,
+        tree_clvm_cost,
+    );
+
+    let (cons_nat_t, cons_clvm_t, cons_nat_c, cons_clvm_c) = time_per_cons_for_list(
+        &mut a,
+        shaprog,
+        atom_nat_c,
+        atom_clvm_c,
+        atom_nat_t,
+        atom_clvm_t,
+        cons_native_time,
+        cons_clvm_time,
+        cons_native_cost,
+        cons_clvm_cost,
+    );
+
+    // taken from benchmark-clvm-cost.rs
+    let cost_scale = ((101094.0 / 39000.0) + (1343980.0 / 131000.0)) / 2.0;
```

Comment:
Since you measure time and cost for the clvm implementation, you meant to compute this scale factor, right? and not use this constant

Reply:
This is removed now for irrelevancy